prog/vm/github_runner.rb

```diff
@@ -220,6 +220,66 @@ def setup_info
     # Remove comments and empty lines before sending them to the machine
     vm.sshable.cmd(command.gsub(/^(\s*# .*)?\n/, ""))
 
+    if github_runner.installation.project.get_ff_transparent_cache
+      hop_setup_forked_runner
+    end
+
+    hop_register_runner
+  end
+
+  label def setup_forked_runner
+    command = <<~COMMAND
+      curl --output actions-runner.tar.gz -L #{Config.github_cache_forked_runner_tarball_uri}
+
+      rm -rf actions-runner
+      mkdir -p actions-runner
+      tar xzf actions-runner.tar.gz -C actions-runner
+
+      sudo chown -R runneradmin:runneradmin ./actions-runner
+
+      cat <<EOT > ./actions-runner/run-withenv.sh
+      #!/bin/bash
+      mapfile -t env </etc/environment
+      exec env -- "${env[@]}" ./actions-runner/run.sh --jitconfig "$1"
+      EOT
+      chmod +x ./actions-runner/run-withenv.sh
+
+      jq '. += [#{setup_info.to_json}]' /imagegeneration/imagedata.json | sudo -u runner tee ./actions-runner/.setup_info
+
+      echo "PATH=$PATH" >> ./actions-runner/.env
+
+      rm -rf /home/runner/actions-runner
+      sudo mv ./actions-runner /home/runner/
+      sudo chown -R runner:runner /home/runner/actions-runner
+
+      echo "CUSTOM_ACTIONS_CACHE_URL=http://localhost:8080/spaceholder/" | sudo tee -a /etc/environment
```

Comment:
Should we use more randomized ports instead of `8080`? Since `8080` is commonly used for development and testing, it might already be part of the customer's workflow